packages/flame/lib/src/components/component_notifier.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+
+/// A [ChangeNotifier] that notifies its listeners when a [Component] is
+/// added or removed, or updated.
+class ComponentNotifier<T extends Component> extends ChangeNotifier {
```

Comment:
Shouldn;t this extend the `Notifiable` mixin? As that would ensure that it will call the add and remove?